src/RescriptEmbedLang.res

```diff
@@ -166,6 +174,24 @@ type handleOtherCommandConfig<'config> = {
 
 type setupConfig = {args: CliArgs.t}
```

Comment:
Initially, I had tinkered with adding a parameter here, which would have avoided the breaking change to `setup`'s return type.
```rescript
type setupConfig  = {
  args: CliArgs.t,
  addWatcher: watcher => unit
}
```
However, I realized that with this API, an embed developer could then start doing funny things with `addWatcher` (e.g. add new watchers after we've already started watching), which would be a whole new can of worms to worry about. By using the return type, we force all watchers to be known once `setup` is done.